### PR TITLE
Update parse/grpc.go in mapper-framework

### PR DESCRIPTION
**Pull request by cl2017**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR Update `grpc.go`:
- Add constant definition
- Optimize the if logic and reduce one if
- Error content added description
- Refine other code details

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: staging/src/github.com/kubeedge/mapper-framework/pkg/util/parse/grpc.go [hunk 1]
```diff
@@ -132,68 +131,67 @@ func buildPropertiesFromGrpc(device *dmiapi.Device) []common.DeviceProperty {
 		var dbconfig common.DBConfig
 		var pushMethod []byte
 		var pushMethodName string
-		if pptv.PushMethod != nil {
-			if pptv.PushMethod.DBMethod != nil {
-				dbMethodName, err = getDBMethodFromGrpc(pptv)
+		if pptv.PushMethod != nil && pptv.PushMethod.DBMethod != nil {
+			dbMethodName, err = getDBMethodFromGrpc(pptv)
```
Comment (wbc6080) [inline staging/src/github.com/kubeedge/mapper-framework/pkg/util/parse/grpc.go:135]: In order to fix the null pointer problem before, we put pptv.PushMethod != nil in a separate if. If you change it like this, if the user does not configure the http or mqtt field of pushMethod, it seems that the null pointer problem will reappear on line 178.

Comment (wbc6080) [reply staging/src/github.com/kubeedge/mapper-framework/pkg/util/parse/grpc.go:135]: What I mean is that after this if, the pushMethodName field will be parsed on line 175. pptv.PushMethod != nil is not judged on line 175. This part of the original code is in the if of pptv.PushMethod != nil.

Comment (wbc6080) [reply staging/src/github.com/kubeedge/mapper-framework/pkg/util/parse/grpc.go:135]: pushmethod

Comment (Shelley-BaoYue) [reply staging/src/github.com/kubeedge/mapper-framework/pkg/util/parse/grpc.go:135]: > What I mean is that after this if, the pushMethodName field will be parsed on line 175. pptv.PushMethod != nil is not judged on line 175. This part of the original code is in the if of pptv.PushMethod != nil.

OK, I get it.

Comment (cl2017) [reply staging/src/github.com/kubeedge/mapper-framework/pkg/util/parse/grpc.go:135]: Oh, yes. But if that's the case, these `if` is confusing, so let me fix it again

**Review by fisherxu — COMMENTED** [review]
/assign @WillardHu 
Plz help take a look :)

**Comment by cl2017** [comment]
> This method `buildPropertiesFromGrpc` is not sensitive to errors and we do not know if it will have an impact on the business. If an error occurs in the method, it will only be logged and properties will be lost.
> 
> In addition, I feel that there are many places in this file that can be abstracted and simplified.

ok, I'm going to look at this function more time, and I'll submit it again later.

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**